File: common/old/mixin/client/render/entity/ZombieBaseEntityRenderer.java
package yummyloop.yummy.old.mixin.client.render.entity;

import kotlin.jvm.internal.Intrinsics;
import net.minecraft.client.render.entity.BipedEntityRenderer;
import net.minecraft.client.render.entity.EntityRenderDispatcher;
import net.minecraft.client.render.entity.feature.FeatureRenderer;
import net.minecraft.client.render.entity.feature.FeatureRendererContext;
import net.minecraft.client.render.entity.model.BipedEntityModel;
import net.minecraft.client.render.entity.model.ZombieEntityModel;
import org.jetbrains.annotations.NotNull;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
import yummyloop.yummy.old.item.armor.render.SpecialArmorFeatureRenderer;

@Mixin({net.minecraft.client.render.entity.ZombieBaseEntityRenderer.class})
public abstract class ZombieBaseEntityRenderer extends BipedEntityRenderer {
    @Inject(
            at = {@At("RETURN")},
            method = "<init>",
            cancellable = true
    )
    private void onZombieBaseEntityRenderer(EntityRenderDispatcher entityRenderDispatcher, ZombieEntityModel zombieEntityModel_1, ZombieEntityModel zombieEntityModel_2, ZombieEntityModel zombieEntityModel_3, CallbackInfo info) {
        this.addFeature((FeatureRenderer)(new SpecialArmorFeatureRenderer((FeatureRendererContext)this, (BipedEntityModel)zombieEntityModel_2, (BipedEntityModel)zombieEntityModel_3)));
    }

    public ZombieBaseEntityRenderer(@NotNull EntityRenderDispatcher entityRenderDispatcher, @NotNull ZombieEntityModel zombieEntityModel_1, @NotNull ZombieEntityModel zombieEntityModel_2, @NotNull ZombieEntityModel zombieEntityModel_3) {
        super(entityRenderDispatcher, (BipedEntityModel)zombieEntityModel_1, 0.5F);
        Intrinsics.checkParameterIsNotNull(entityRenderDispatcher, "entityRenderDispatcher");
        Intrinsics.checkParameterIsNotNull(zombieEntityModel_1, "zombieEntityModel_1");
        Intrinsics.checkParameterIsNotNull(zombieEntityModel_2, "zombieEntityModel_2");
        Intrinsics.checkParameterIsNotNull(zombieEntityModel_3, "zombieEntityModel_3");
    }
}

File: src/main/yummyloop/example/mixin/render/entity/ZombieBaseEntityRenderer.kt
package yummyloop.example.mixin.render.entity

import net.minecraft.client.render.entity.BipedEntityRenderer
import net.minecraft.client.render.entity.EntityRenderDispatcher
import net.minecraft.client.render.entity.model.ZombieEntityModel
import net.minecraft.entity.mob.ZombieEntity
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Inject
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo
import yummyloop.example.item.armor.render.SpecialArmorFeatureRenderer
import net.minecraft.client.render.entity.ZombieBaseEntityRenderer as VanillaZombieBaseEntityRenderer

@Mixin(VanillaZombieBaseEntityRenderer::class)
abstract class ZombieBaseEntityRenderer<T : ZombieEntity, M : ZombieEntityModel<T>>
constructor(entityRenderDispatcher : EntityRenderDispatcher, zombieEntityModel_1 : M, zombieEntityModel_2 : M, zombieEntityModel_3 : M) :
        BipedEntityRenderer<T, M> (entityRenderDispatcher, zombieEntityModel_1,0.5f) {

    @Inject(at = [At("RETURN")], method = ["<init>"], cancellable = true)
    private fun onZombieBaseEntityRenderer(entityRenderDispatcher : EntityRenderDispatcher, zombieEntityModel_1 : M, zombieEntityModel_2 : M, zombieEntityModel_3 : M, info: CallbackInfo) {
        // Special armor feature renderer
        this.addFeature(SpecialArmorFeatureRenderer(this, zombieEntityModel_2, zombieEntityModel_3))
    }
}
